### PR TITLE
Adds x-pack for monitoring

### PR DESCRIPTION
**Pull request by marcinc**
Enables Elastic X-Pack. There is 30-day trial license installed by default. After it expires it'll downgrade most of X-Pack features unless we decide to get the licence. Free license includes basic monitoring though and this is what we're after at the moment. I've disabled trial security plugin as it requires quite a lot of configuration and should be evaluated before we use it.

### CODE REVIEW

**Comment by stefancocora** [comment]
Would you envisage using x-pack "actions" to send alerts to sysdig which then sends to slack/opsgenie to trigger alerts ?

**Comment by marcinc** [comment]
@stefancocora We need to evaluate advanced features of x-pack first and decide whether it's worth it or not. Initially we're interested in basic monitoring.

File: config/elasticsearch.yml [hunk 1]
```diff
@@ -37,3 +37,6 @@ searchguard.ssl.transport.truststore_alias: ca
 # searchguard.ssl.http.keystore_alias: cert
 # searchguard.ssl.http.truststore_filepath: certs/truststore.jks
 # searchguard.ssl.http.truststore_alias: ca
+
+# TODO: Evaluate whether it's worth paying for X-Pack advanced features. Basic monitoring is free.
+xpack.security.enabled: false
```
Comment (vaijab) [inline config/elasticsearch.yml:42]: How is this related to advanced features of X-Pack?

File: Dockerfile [hunk 2]
```diff
@@ -10,6 +10,8 @@ ENV ES_VERSION 5.1.1
 RUN wget -q https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-${ES_VERSION}.tar.gz -O - | tar -xzf -; mv elasticsearch-${ES_VERSION} /elasticsearch && mkdir -p /elasticsearch/config/scripts /elasticsearch/plugins
 RUN /elasticsearch/bin/elasticsearch-plugin install io.fabric8:elasticsearch-cloud-kubernetes:5.1.1
 RUN /elasticsearch/bin/elasticsearch-plugin install com.floragunn:search-guard-ssl:5.1.1-19 && (cd /elasticsearch/plugins/search-guard-ssl/ && wget -q http://repo1.maven.org/maven2/io/netty/netty-tcnative/1.1.33.Fork24/netty-tcnative-1.1.33.Fork24-linux-x86_64-fedora.jar)
+RUN /elasticsearch/bin/elasticsearch-plugin install x-pack && \
+    chmod -R +rx /elasticsearch/config/x-pack
```
Comment (vaijab) [inline Dockerfile]: What is in config/x-pack directory that it `+x` is required?

File: config/elasticsearch.yml (hunk 1 above)
Comment (marcinc) [reply config/elasticsearch.yml:42]: Security module is an advanced feature in x-pack. Security will be disabled in 30 days unless we get the license so I disabled it to start with. If enabled it'd require extra configuration which we aren't sure we gonna be using. The only one which remain free to use is monitoring.

File: Dockerfile (hunk 2 above)
Comment (marcinc) [reply Dockerfile]: `/elasticsearch/config/x-pack` needs `+x` otherwise:

```Exception in thread "main" org.elasticsearch.bootstrap.BootstrapException: java.nio.file.AccessDeniedException: /elasticsearch/config/x-pack/users_roles
Likely root cause: java.nio.file.AccessDeniedException: /elasticsearch/config/x-pack/users_roles
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:84)
```

Individual config files don't need +x but still need +r so can split that in two separate lines.

File: config/elasticsearch.yml (hunk 1 above)
Comment (vaijab) [reply config/elasticsearch.yml:42]: Gotcha.

**Review by vaijab — APPROVED** [review]
Thanks.
